instrumentation/jmx-metrics/javaagent/src/main/resources/jmx/rules/wildfly.yaml

```diff
@@ -25,47 +25,49 @@ rules:
         unit: ns
       errorCount:
   - bean: jboss.as:subsystem=undertow,server=*,http-listener=*
+    metricPrefix: wildfly.
     metricAttribute:
       server: param(server)
       listener: param(http-listener)
     type: counter
     unit: By
     mapping:
       bytesSent:
-        metric: wildfly.network.io
+        metric: network.io
         desc: Total number of bytes transferred
         metricAttribute:
-          direction: const(out)
+          network.io.direction: const(transmit)
       bytesReceived:
-        metric: wildfly.network.io
+        metric: network.io
         desc: Total number of bytes transferred
         metricAttribute:
-          direction: const(in)
+          network.io.direction: const(receive)
   - bean: jboss.as:subsystem=datasources,data-source=*,statistics=pool
-    unit: "1"
+    unit: "{connection}"
     metricAttribute:
       data_source: param(data-source)
     mapping:
       ActiveCount:
-        metric: wildfly.db.client.connections.usage
+        metric: wildfly.db.client.connection.count
         metricAttribute:
-          state: const(used)
+          db.client.connection.state: const(used)
```

Comment:
Unless I'm missing something, the metric will have a metric attribute which will always have the same constant value. What is really the point of having such an attribute?

Reply:
You are correct here, having a constant metric attribute only makes sense to provide breakdown of the same metric, occurences of `wildfly.db.client.connection.usage` should be renamed to `wildfly.db.client.connection.count`. I am currently trying to validate with the implementation if there is an overlap between the connection states or if it's an effective partition (in which case we can provide a breakdown).